src/dmd/env.d

```diff
@@ -0,0 +1,134 @@
+module dmd.env;
+
+import core.stdc.string;
+import core.sys.posix.stdlib;
+import dmd.globals;
+import dmd.root.array;
+import dmd.root.rmem;
+
+version (Windows) extern (C) int putenv(const char*) nothrow;
+
+/**
+Used to store an environment variable including its value.
+*/
+struct Env
+{
+    private const(char)* setString; /// null-terminated string of the form "VAR=VALUE"
```

Comment:
Can we use a D string?

Reply:
We could but I don't see the benefit here.

Reply:
Right but this isn't being used as a string.  It's just being passed to the C function `putenv`.

Reply:
I could still make it a string, but this isn't really being used like other strings we want to convert to D strings, hence why I don't see the benefit.  Maybe you can show me a place where using a D string would prevent an error?  I don't see any.

Reply:
This member MUST be a null-terminated string.  And I've noticed that when DMD has to use a null-terminated string, it usually uses `char*` to signify that.  If I use `char[]` would that be confusing?

Reply:
Ok I've noticed other places where DMD uses D arrays for strings even when they need to be null-terminated.  So I've gone ahead and made it a `string`.